source/common/http/header_utility.cc

```diff
@@ -190,17 +190,16 @@ bool HeaderUtility::schemeIsValid(const absl::string_view scheme) {
 }
 
 bool HeaderUtility::headerValueIsValid(const absl::string_view header_value) {
-  return nghttp2_check_header_value(reinterpret_cast<const uint8_t*>(header_value.data()),
-                                    header_value.size()) != 0;
+  return http2::adapter::HeaderValidator::IsValidHeaderValue(header_value,
+                                                             http2::adapter::ObsTextOption::kAllow);
 }
 
 bool HeaderUtility::headerNameContainsUnderscore(const absl::string_view header_name) {
   return header_name.find('_') != absl::string_view::npos;
 }
 
 bool HeaderUtility::authorityIsValid(const absl::string_view header_value) {
-  return nghttp2_check_authority(reinterpret_cast<const uint8_t*>(header_value.data()),
-                                 header_value.size()) != 0;
+  return http2::adapter::HeaderValidator::IsValidAuthority(header_value);
```

Comment:
We've confirmed this is identical to nghttp as well?

Reply:
Looks like the QUICHE method rejects "@", which nghttp2 allows. I think this is slightly more adherent to [RFC 3986 section 3.2.2](https://www.rfc-editor.org/rfc/rfc3986#section-3.2.2).

Reply:
Because it's stricter, I think that means we should have a runtime guard for this behavior so that someone can revert it.

Reply:
Agreed. @birenroy can you add a guard for this?

Reply:
Done.